platform/android/src/jni.cpp

```diff
@@ -1620,6 +1753,363 @@ void JNICALL nativeRemoveCustomLayer(JNIEnv *env, jobject obj, jlong nativeMapVi
     nativeMapView->getMap().removeCustomLayer(std_string_from_jstring(env, id));
 }
 
+// Offline calls begin
+
+jlong JNICALL createDefaultFileSource(JNIEnv *env, jobject obj, jstring cachePath_, jstring assetRoot_, jlong maximumCacheSize) {
+    mbgl::Log::Debug(mbgl::Event::JNI, "createDefaultFileSource");
+    std::string cachePath = std_string_from_jstring(env, cachePath_);
+    std::string assetRoot = std_string_from_jstring(env, assetRoot_);
+    mbgl::DefaultFileSource *defaultFileSource = new mbgl::DefaultFileSource(cachePath, assetRoot, maximumCacheSize);
+    jlong defaultFileSourcePtr = reinterpret_cast<jlong>(defaultFileSource);
+    return defaultFileSourcePtr;
+}
+
+void JNICALL setAccessToken(JNIEnv *env, jobject obj, jlong defaultFileSourcePtr, jstring accessToken_) {
+    mbgl::Log::Debug(mbgl::Event::JNI, "setAccessToken");
+    assert(defaultFileSourcePtr != 0);
+    std::string accessToken = std_string_from_jstring(env, accessToken_);
+    mbgl::DefaultFileSource *defaultFileSource = reinterpret_cast<mbgl::DefaultFileSource *>(defaultFileSourcePtr);
+    defaultFileSource->setAccessToken(accessToken);
+}
+
+jstring JNICALL getAccessToken(JNIEnv *env, jobject obj, jlong defaultFileSourcePtr) {
+    mbgl::Log::Debug(mbgl::Event::JNI, "getAccessToken");
+    assert(defaultFileSourcePtr != 0);
+    mbgl::DefaultFileSource *defaultFileSource = reinterpret_cast<mbgl::DefaultFileSource *>(defaultFileSourcePtr);
+    std::string accessToken = defaultFileSource->getAccessToken();
+    return std_string_to_jstring(env, accessToken);
+}
+
+void JNICALL listOfflineRegions(JNIEnv *env, jobject obj, jlong defaultFileSourcePtr, jobject listCallback) {
+    mbgl::Log::Debug(mbgl::Event::JNI, "listOfflineRegions");
+
+    // Checks
+    assert(defaultFileSourcePtr != 0);
+    if (listCallback == nullptr) {
+        mbgl::Log::Error(mbgl::Event::JNI, "Callback has to be set.");
+        return;
+    }
+
+    // Makes sure the callback doesn't get GC'ed
+    listCallback = reinterpret_cast<jobject>(env->NewGlobalRef(listCallback));
```

Comment:
Ready for your :eyes: in https://github.com/mapbox/mapbox-gl-native/commit/2fe6aa121c80f6d977800b14416ddca1b2d95c27.